### PR TITLE
Move 4722 (dangling symlink) to misc tests, remove dangling symlink from repo

### LINKED ISSUE #7065: Get rid of dangiling symlink in repository
#### Version

8.7

#### Operating system
Linux

#### Description of the problem

The Coq git currently contains a dangling symlink: `test-suite/bugs/closed/4722/tata`. This triggers warnings in some tools, e.g. opam when pinning the Coq dir (it uses rsync which complains about the bad symlink), or gitlab CI caches.

The symlink is a test for #4722. This test could be moved to a misc test (suggested by @SkySkimmer) so that instead of storing a dangling symlink in the repo, we create it at test time.

### PR DESCRIPTION
**Pull request by RalfJung**
**Kind:** bug fix

Fixes #7065

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Should we remove the link after running coqtop? It seems like if some tool has issues with it, the issues will remain if the tool runs the test suite.

File: test-suite/misc/4722.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+set -e
+
+if ! test -d misc/4722; then
```
Comment (SkySkimmer) [inline test-suite/misc/4722.sh]: I don't like this test for some reason. Would you mind doing 
```bash
mkdir -p misc/4722
ln -sfT toto misc/4722/tata
touch misc/4722.v
```
instead?

**Review by SkySkimmer — APPROVED** [review]
>Like so?

Yes but please squash.

File: test-suite/misc/4722.sh (hunk 1 above)
Comment (RalfJung) [reply test-suite/misc/4722.sh]: *shrug* I find that entirely unreadable and much less clear than my code, but I don't care strongly.

**Comment by RalfJung** [comment]
> Yes but please squash.

Done.

File: test-suite/misc/4722.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+set -e
+
+# create test files
+mkdir -p misc/4722
+ln -sfT toto misc/4722/tata
```
Comment (SkySkimmer) [inline test-suite/misc/4722.sh]: Sorry, it seems `ln -T` doesn't exist on OSX.
It shouldn't be necessary in this case (tata should never be a directory) so just remove it.

**Review by SkySkimmer — CHANGES_REQUESTED** [review]

File: test-suite/misc/4722.sh (hunk 2 above)
Comment (RalfJung) [reply test-suite/misc/4722.sh]: Fixed.

**Review by SkySkimmer — APPROVED** [review]

**Pull request merged**